dataland-internal-storage/src/main/kotlin/db/migration/V19__MigrateEutaxonomyNonFinancialsExtendedDatapoints.kt

```diff
@@ -0,0 +1,117 @@
+package db.migration
+
+import db.migration.utils.DataTableEntity
+import db.migration.utils.getOrJavaNull
+import db.migration.utils.migrateCompanyAssociatedDataOfDatatype
+import org.flywaydb.core.api.migration.BaseJavaMigration
+import org.flywaydb.core.api.migration.Context
+import org.json.JSONArray
+import org.json.JSONObject
+
+/**
+ * This migration script updates the existing EU taxonomy non-financials datasets and migrates all
+ * existing BaseDataPoints to ExtendedDataPoints.
+ */
+class V19__MigrateEutaxonomyNonFinancialsExtendedDatapoints : BaseJavaMigration() {
+
+    private val relevantFields = listOf(
+        "scopeOfEntities",
+        "nfrdMandatory",
+        "euTaxonomyActivityLevelReporting",
+        "numberOfEmployees",
+        "enablingShareInPercent",
+        "transitionalShareInPercent",
+        "relativeShareInPercent",
+        "substantialContributionToClimateChangeMitigationInPercent",
+        "substantialContributionToClimateChangeAdaptationInPercent",
+        "substantialContributionToSustainableUseAndProtectionOfWaterAndMarineResourcesInPercent",
+        "substantialContributionToTransitionToACircularEconomyInPercent",
+        "substantialContributionToPollutionPreventionAndControlInPercent",
+        "substantialContributionToProtectionAndRestorationOfBiodiversityAndEcosystemsInPercent",
+        "absoluteShare",
+        "fiscalYearEnd",
+        "fiscalYearDeviation",
+        "alignedActivities",
+        "nonAlignedActivities",
+    )
+
+    /**
+     * Create a nested JSON object from a JSON object and a key.
+     * @param jsonObject JSON object
+     * @param key key corresponding to the JSON object
+     */
+    private fun createNestedJsonObject(jsonObject: JSONObject, key: String) {
+        val nestedJson = JSONObject()
+        nestedJson.put("value", jsonObject.getOrJavaNull(key))
+        jsonObject.put(key, nestedJson)
+    }
+
+    /**
+     * Check if the keys of a JSON object are relevant fields and, if so, created a nested JSON object.
+     * @param jsonObject JSON object
+     */
+    private fun checkForRelevantFieldsInJsonObjectKeys(jsonObject: JSONObject) {
+        jsonObject.keys().forEach {
+            if (it == "absoluteShare") {
+                val absoluteShare = jsonObject[it] as JSONObject
+                val amount = absoluteShare["amount"]
+                absoluteShare.remove("amount")
+                absoluteShare.put("value", amount)
+            } else if (it in relevantFields) {
+                createNestedJsonObject(jsonObject, it)
+            } else {
+                // Do nothing
```

Comment:
remove `else` ?

Reply:
Is marked as a change because I did not merge main into branch earlier